### PR TITLE
Rails guide

### PR DESCRIPTION
**Pull request by CrowdHailer**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -17,18 +17,16 @@ Incorporating DID.app gives your users a fast, easy, and safe way to sign in usi
...
-
-<!-- New to DID.app? Follow one of our implementation guides
+Or try one of our specific platform guides.
 
 - [Phoenix & Elixir](/guides/phoenix-elixir-openid-connect-integration)
 - [Express & Node.js](/guides/express-nodejs-openid-connect-integration)
 - [Wordpress](/guides/wordpress-openid-connect-client-plugin-integration/)
 
-Can't find what your platform, the simplest way to integrate with DID is using an OpenID Connect library for your platform.
+<!-- Can't find what your platform, the simplest way to integrate with DID is using an OpenID Connect library for your platform.
```
Comment (richardesigns) [inline docs/index.md]:
```suggestion
<!-- Can't find your platform? The simplest way to integrate with DID is using an OpenID Connect library for your platform.
```

File: guides/ruby-on-rails-openid-connect-integration/index.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+Rails.application.routes.draw do
+  get 'welcome/index'
+  root 'welcome#index'
+end
+```
+
+
+### Outline of Authentication Flow
+
+Authenticating End Users takes two steps.
```
Comment (richardesigns) [inline guides/ruby-on-rails-openid-connect-integration/index.md]:
```suggestion
Authenticating end users takes two steps.
```

File: guides/ruby-on-rails-openid-connect-integration/index.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  <button type="submit">Sign in</button>
+</form>
+```
+
+The `CLIENT_ID` is a value provided to you by DID.app that we will generate later.  
+The `callback_url` refers to a controller action that we will implement in the next section.
+
+### Create a session controller to handle authentication callback
+
+Once a user has authenticated, with DID.app, they will be redirected to the url that was specified in the authentication request.
```
Comment (richardesigns) [inline guides/ruby-on-rails-openid-connect-integration/index.md]:
```suggestion
Once a user has authenticated with DID.app, they will be redirected to the url that was specified in the authentication request.
```

File: guides/ruby-on-rails-openid-connect-integration/index.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+Rails.application.routes.draw do
+  # Other routes
+  # ...
+
+  get '/session/callback', to: 'session#callback', as: 'callback'
+end
+```
+
+Create a new file for a session controller.
+Add the following code to implement the callback route.
```
Comment (richardesigns) [inline guides/ruby-on-rails-openid-connect-integration/index.md]:
```suggestion
Add the following code to implement the callback route:
```

File: guides/ruby-on-rails-openid-connect-integration/index.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    )
+
+    session[:current_user_id] = data["userinfo"]["sub"]
+    redirect_to root_path
+  end
+end
+```
+
+The redirect url back to your app will include a query parameter containing a one-time code.
+To finish authenticating a user this code needs to be exchanged for a token and user information.
```
Comment (richardesigns) [inline guides/ruby-on-rails-openid-connect-integration/index.md]:
```suggestion
To finish authenticating a user, this code needs to be exchanged for a token and user information.
```

File: guides/ruby-on-rails-openid-connect-integration/index.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+From this user information we extract the `sub` parameter, short for `subject`, this parameter is a stable identifier for the user.
```
Comment (richardesigns) [inline guides/ruby-on-rails-openid-connect-integration/index.md]:
```suggestion
From this user information we extract the `sub` parameter which is short for `subject`. This parameter is a stable identifier for the user.
```

File: guides/ruby-on-rails-openid-connect-integration/index.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+A verified email address is also returned as part of the user information.
+
+This request will require your `CLIENT_ID` and `CLIENT_SECRET`.
+It is good practice to keep these values out of your source code and so we are using environment variables to configure them.
```
Comment (richardesigns) [inline guides/ruby-on-rails-openid-connect-integration/index.md]:
```suggestion
It is good practice to keep these values out of your source code so we are using environment variables to configure them.
```

File: guides/ruby-on-rails-openid-connect-integration/index.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+### Display Sign Out Button to Authenticated User
+
+Once a user has authenticated we no longer want to show them a sign in button.
```
Comment (richardesigns) [inline guides/ruby-on-rails-openid-connect-integration/index.md]:
```suggestion
Once a user has authenticated, we no longer want to show them a sign in button.
```

File: guides/ruby-on-rails-openid-connect-integration/index.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+Probably the most helpful thing to show instead would be a sign out button.
+
+```erb
+<% if session[:current_user_id] %>
+  <%= link_to 'Sign out', sign_out_path %>
+<% else %>
+  <!-- Sign in form from earlier -->
```
Comment (richardesigns) [inline guides/ruby-on-rails-openid-connect-integration/index.md]: Any reason why we wouldn't just put the form in again here?  If there is a reason can we link to an anchor on the page?  Not a blocker just would be nice.

File: guides/ruby-on-rails-openid-connect-integration/index.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+  end
+end
+```
+
+Signing out a user is done by clearing the session information.
+
+### Get App Identifiers DID.app
+
+The final step is to get a `CLIENT_ID` and `CLIENT_SECRET` for you app.
+You will need a account, [Sign up]({{ site.auth_origin }}/sign_up) to create one.
```
Comment (richardesigns) [inline guides/ruby-on-rails-openid-connect-integration/index.md]:
```suggestion
You will need an account, [Sign up]({{ site.auth_origin }}/sign_up) to create one.
```

File: docs/index.md (hunk 1 above)
Comment (CrowdHailer) [reply docs/index.md]: You probably can ignore the commented out stuff

File: guides/ruby-on-rails-openid-connect-integration/index.md (hunk 9 above)
Comment (CrowdHailer) [reply guides/ruby-on-rails-openid-connect-integration/index.md:131]: Just though this was easier for focusing on what is new for this part of the guide

**Pull request merged**